sources/code/main/modules/extensions.ts

```diff
@@ -31,32 +38,30 @@ async function parseImports(cssString: string, importCalls:string[], maxTries=5)
       promises.push(Promise.reject(new Error("Circular reference in CSS imports are disallowed!")));
       return;
     }
-    importCalls.push(file);
-    promises.push(fetchOrRead(file)
+    promises.push(fetchOrRead(file, prevImport)
       .then(data => {
         if (data.download)
           return data.download.then(data => data.text());
         else
           return data.read.then(data => data.toString());
       })
+      .then(content => parseImports(content, [...importCalls, file], maxTries, dirname(file)))
       .then(content => cssString = cssString.replace(singleImport, content))
     );
   });
   try {
     await Promise.all(promises);
   } catch(error) {
     if(maxTries > 0) {
-      console.warn("Couldn't resolve CSS theme imports, retrying again...");
-      maxTries--;
+      console.error(error);
```

Comment:
Please remove the `console.error` call here. I don't want to print any errors when the code when there's no failure yet and the overall algorithm tries to take care of an error. We already inform the user there's some kind of the re-approach, I think that's more than enough, at least for the production.

Reply:
Good catch, I didn't mean to commit this, it was just for finding the `importCalls` problem mentioned earlier.